dbt/include/snowflake/macros/materializations/incremental.sql

```diff
@@ -20,15 +20,15 @@
        Otherwise, play it safe by using a temporary table.
   #} */
 
-  {% if language != "sql" %}
+  {% if language == "python" %}
     {{ return("table") }}
-  {% elif strategy in ("default", "merge") and merge_tmp_relation_type == "table" %}
+  {% elif tmp_relation_type == "table" %}
     {{ return("table") }}
-  {% elif strategy in ("default", "merge") and merge_tmp_relation_type == "view" %}
+  {% elif tmp_relation_type == "view" %}
     {{ return("view") }}
   {% elif strategy in ("default", "merge", "append") %}
     {{ return("view") }}
-  {% elif strategy == "delete+insert" and unique_key is none %}
+  {% elif unique_key is none %}
```

Comment:
Could we change the minimal number of lines to preserve as much of the logic we worked on as possible?

Something like:
```sql
  {% if language != "sql" %}
    {{ return("table") }}
  {% elif tmp_relation_type == "table" %}
    {{ return("table") }}
  {% elif tmp_relation_type == "view" %}
    {{ return("view") }}
  {% elif strategy in ("default", "merge", "append") %}
    {{ return("view") }}
  {% elif strategy == "delete+insert" and unique_key is none %}
```

Reply:
sure can convert back some, will wind up with a few extra lines through to incorporate the conditions for the exceptions we wish to raise

Reply:
The exceptions can probably be in a separate block prior to this.

Something like:
```sql
# Handle un-happy path first
if language == 'python' and incremental_tmp_relation_type == 'view':
     raise exception
elif incremental_strategy == 'delete+insert' and incremental_tmp_relation_type == 'table':
     raise exception

# Handle happy path second
if language != "sql":
    return "table"
elif ...
```

Reply:
sounds good, will add it now, have we decided on the config name? `tmp_relation_type` or `incremental_tmp_relation_type` btw?

Reply:
My preference is `tmp_relation_type`, and I didn't see any arguments to the contrary [here](https://github.com/dbt-labs/dbt-snowflake/issues/402#issuecomment-1387300212).